### PR TITLE
update go to 1.21 

### PR DESCRIPTION
**Pull request by lhy1024**
1. update golang to 1.21
2. update mockery to v2.40.3
3. update go-lint to 1.59.1
4. fix some lint 
   - unused-parameter
   - SA1019: reflect.StringHeader has been deprecated
   - dot-imports
5. remove depguard temporary

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -25,12 +21,16 @@ issues:
...
+    - swaggerspec
+    - pkg/uiserver
+    - ui
 
 linters:
   enable:
     # Additionally enable some checkers
     - asciicheck
-    - depguard
+    # - depguard
```
Comment (lhy1024) [inline .golangci.yml:33]: I close it because I am not sure how tidb-dashboard will do about it.

File: pkg/keyvisual/matrix/key.go [hunk 2]
```diff
@@ -29,7 +27,5 @@ func (km *KeyMap) SaveKeys(keys []string) {
 }
 
 func equal(keyA, keyB string) bool {
```
Comment (lhy1024) [inline pkg/keyvisual/matrix/key.go:29]: replace it

File: pkg/keyvisual/region/utils.go [hunk 3]
```diff
@@ -3,31 +3,15 @@
...
-func String(b []byte) (s string) {
-	if len(b) == 0 {
-		return ""
-	}
-	pbytes := (*reflect.SliceHeader)(unsafe.Pointer(&b))   // #nosec
-	pstring := (*reflect.StringHeader)(unsafe.Pointer(&s)) // #nosec
-	pstring.Data = pbytes.Data
-	pstring.Len = pbytes.Len
-	return
+func String(b []byte) string {
```
Comment (lhy1024) [inline pkg/keyvisual/region/utils.go:10]: ditto

File: pkg/keyvisual/region/utils.go [hunk 4]
```diff
@@ -3,31 +3,15 @@
...
-	if len(s) == 0 {
-		return
-	}
-	pbytes := (*reflect.SliceHeader)(unsafe.Pointer(&b))   // #nosec
-	pstring := (*reflect.StringHeader)(unsafe.Pointer(&s)) // #nosec
-	pbytes.Data = pstring.Data
-	pbytes.Len = pstring.Len
-	pbytes.Cap = pstring.Len
-	return
+func Bytes(s string) []byte {
```
Comment (lhy1024) [inline pkg/keyvisual/region/utils.go:15]: ditto

**Review by nolouch — APPROVED** [review]
lgtm.
PTAL @baurine @Renkai

**Review by baurine — APPROVED** [review]

**Comment by baurine** [comment]
@lhy1024 please sign the CLA, thanks~

**Pull request merged**